clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesV2ClusterProvider.java

```diff
@@ -256,7 +256,7 @@ public boolean supportsMinimalClusters() {
       serverGroupToServerGroupManagerKeys.put(
           serverGroupDatum.getId(),
           kindMap.translateSpinnakerKind(SERVER_GROUP_MANAGERS).stream()
-              .map(kind -> serverGroupDatum.getRelationships().get(kind.toString()))
+              .map(kind -> serverGroupDatum.getRelationships().get(kind.getName()))
```

Comment:
Looking at the code more closely, I'm not sure that we should need to make this change.

It looks like the code to store the relationships in the cache is calling `getGroup` on the key:
https://github.com/spinnaker/clouddriver/blob/cfaec246181f207fa46c906d9461c4ae81a6ed6f/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCacheData.java#L118

And `getGroup` is itself calling `toString` on the `kind`:
https://github.com/spinnaker/clouddriver/blob/cfaec246181f207fa46c906d9461c4ae81a6ed6f/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/Keys.java#L221

So I think we should be using `kind.toString()` as they key when we store the relationships in the cache, and thus should also use that when retrieving.

Reply:
@ezimanyi agreed. This may not be the best way to fix it. I was not able to figure out the reason this is happening. So for now i have reverted the second fix. Hope you can approve the first fix ( which should be the only one present now)

Reply:
Sounds good, we can merge this, and then separately figure out why the cache is not including the fully-scoped name.  Thanks!